### PR TITLE
gRPC interface implementation for the discord and email provider

### PR DESCRIPTION
**Pull request by ThomasCardin**
- Implemented the `SendMessage` function that is going to communicate with the client

### CODE REVIEW

**Comment by ThomasCardin** [comment]
@cguertin14

File: cmd/discord-provider/main.go [hunk 1]
```diff
@@ -1 +1,39 @@
...
+
+type Server struct {
+	discord_provider.DiscordProviderServer
+}
+
+func (server *Server) SendMessage(ctx context.Context, message *discord_provider.SendMessageRequest) (*discord_provider.SendMessageResponse, error) {
+	log.Printf("Received message content from client: %s", message.Sender)
+	// Sends a message to a Discord webhook here
+	return &discord_provider.SendMessageResponse{}, nil
+}
```
Comment (cguertin14) [inline cmd/discord-provider/main.go:22]: this part should be in a separate Go file, same for other gRPC servers

File: cmd/email-provider/main.go [hunk 2]
```diff
@@ -1 +1,39 @@
...
+
+type Server struct {
+	email_provider.EmailProviderServer
+}
+
+func (server *Server) SendMessage(ctx context.Context, message *email_provider.SendEmailRequest) (*email_provider.SendEmailResponse, error) {
+	log.Printf("Received message content from client: %s", message.Sender)
+	// sends an email to someone here
+	return &email_provider.SendEmailResponse{}, nil
+}
```
Comment (cguertin14) [inline cmd/email-provider/main.go:22]: same here

**Review by cguertin14 — CHANGES_REQUESTED** [review]

**Comment by ThomasCardin** [comment]
Thanks for the review @cguertin14! I will check this out ASAP.

**Comment by cguertin14** [comment]
> Thanks for the review @cguertin14! I will check this out ASAP.

No rush 🤘🏻

**Review by cguertin14 — APPROVED** [review]

**Pull request merged**